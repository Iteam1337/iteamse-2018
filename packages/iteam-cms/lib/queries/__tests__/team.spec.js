const resolver = require('../team')

describe('queries/team', () => {
  let context

  beforeEach(() => {
    context = {
      getByContentType: jest.fn().mockResolvedValue([
        {
          sys: {
            space: {
              sys: {
                type: 'Link',
                linkType: 'Space',
                id: 'rj4r6yfcesw5',
              },
            },
            id: '6FvzqbEEcE2IoAyiGAU4gE',
            type: 'Entry',
            createdAt: '2017-06-28T11:59:14.423Z',
            updatedAt: '2017-08-31T09:03:32.232Z',
            revision: 2,
            contentType: {
              sys: {
                type: 'Link',
                linkType: 'ContentType',
                id: 'teamMembers',
              },
            },
            locale: 'sv-SE',
          },
          fields: {
            name: 'Johanna Månsson Grahn',
            title: 'Samarbete & rekrytering',
            short: 'jmn',
            avatar: {
              sys: {
                space: {
                  sys: {
                    type: 'Link',
                    linkType: 'Space',
                    id: 'rj4r6yfcesw5',
                  },
                },
                id: '64biXO1wJ2MA6wqQqWsI4o',
                type: 'Asset',
                createdAt: '2017-08-31T09:03:16.638Z',
                updatedAt: '2017-08-31T09:03:16.638Z',
                revision: 1,
                locale: 'sv-SE',
              },
              fields: {
                title: 'jmn',
                file: {
                  url:
                    '//images.ctfassets.net/rj4r6yfcesw5/64biXO1wJ2MA6wqQqWsI4o/2e3876889e13a9a1750c789b2b1e2a07/jmn.jpg',
                  details: {
                    size: 16518,
                    image: {
                      width: 225,
                      height: 225,
                    },
                  },
                  fileName: 'jmn.jpg',
                  contentType: 'image/jpeg',
                },
              },
            },
            location: 'Göteborg',
            email: 'johanna.grahn@iteam.se',
          },
        },
        {
          sys: {
            space: {
              sys: {
                type: 'Link',
                linkType: 'Space',
                id: 'rj4r6yfcesw5',
              },
            },
            id: '4tyjybv5Yke2MskskygOa0',
            type: 'Entry',
            createdAt: '2017-06-28T11:57:40.999Z',
            updatedAt: '2017-08-31T09:02:00.223Z',
            revision: 3,
            contentType: {
              sys: {
                type: 'Link',
                linkType: 'ContentType',
                id: 'teamMembers',
              },
            },
            locale: 'sv-SE',
          },
          fields: {
            name: 'Hans Rollman',
            title: 'Samarbete',
            short: 'hrn',
            avatar: {
              sys: {
                space: {
                  sys: {
                    type: 'Link',
                    linkType: 'Space',
                    id: 'rj4r6yfcesw5',
                  },
                },
                id: '2uQtyFGphOCg4U6e2mm2oY',
                type: 'Asset',
                createdAt: '2017-08-31T09:01:43.832Z',
                updatedAt: '2017-08-31T09:01:43.832Z',
                revision: 1,
                locale: 'sv-SE',
              },
              fields: {
                title: 'hrn',
                file: {
                  url:
                    '//images.ctfassets.net/rj4r6yfcesw5/2uQtyFGphOCg4U6e2mm2oY/c433294b7e3f0dfe9becd87c1c9907d3/hrn.jpg',
                  details: {
                    size: 14435,
                    image: {
                      width: 225,
                      height: 225,
                    },
                  },
                  fileName: 'hrn.jpg',
                  contentType: 'image/jpeg',
                },
              },
            },
            location: 'Stockholm',
            email: 'hans.rollman@iteam.se',
          },
        },
        {
          sys: {
            space: {
              sys: {
                type: 'Link',
                linkType: 'Space',
                id: 'rj4r6yfcesw5',
              },
            },
            id: '2eap07OhAYIaU00kEMaSqs',
            type: 'Entry',
            createdAt: '2018-03-23T10:40:35.562Z',
            updatedAt: '2018-03-24T07:37:49.839Z',
            revision: 2,
            contentType: {
              sys: {
                type: 'Link',
                linkType: 'ContentType',
                id: 'teamMembers',
              },
            },
            locale: 'sv-SE',
          },
          fields: {
            name: 'Rickard Laurin',
            title: 'Webbutvecklare',
            short: 'rln',
            location: 'Stockholm',
            email: 'rickard.laurin@iteam.se',
          },
        },
        {
          sys: {
            space: {
              sys: {
                type: 'Link',
                linkType: 'Space',
                id: 'rj4r6yfcesw5',
              },
            },
            id: '6xrssgiXu0amGy4Qao22YS',
            type: 'Entry',
            createdAt: '2017-06-28T11:58:24.556Z',
            updatedAt: '2018-03-24T07:37:58.893Z',
            revision: 3,
            contentType: {
              sys: {
                type: 'Link',
                linkType: 'ContentType',
                id: 'teamMembers',
              },
            },
            locale: 'sv-SE',
          },
          fields: {
            name: 'Maria Sölscher',
            title: 'Rekrytering',
            short: 'msr',
            location: 'Stockholm',
            email: 'maria.solscher@iteam.se',
          },
        },
        {
          sys: {
            space: {
              sys: {
                type: 'Link',
                linkType: 'Space',
                id: 'rj4r6yfcesw5',
              },
            },
            id: '5jm03jKTNSAoCwyAI6q6eA',
            type: 'Entry',
            createdAt: '2017-06-28T12:00:17.501Z',
            updatedAt: '2018-03-24T07:37:40.913Z',
            revision: 4,
            contentType: {
              sys: {
                type: 'Link',
                linkType: 'ContentType',
                id: 'teamMembers',
              },
            },
            locale: 'sv-SE',
          },
          fields: {
            name: 'Mats Egardt',
            title: 'Drift & support',
            short: 'met',
            location: 'Stockholm',
            email: 'mats.egardt@iteam.se',
          },
        },
      ]),
    }
  })

  it('should call getContentType with correct content type', async () => {
    try {
      await resolver(null, {}, context)

      expect(context.getByContentType).toHaveBeenCalledWith('teamMembers')
    } catch (error) {
      throw new Error(error)
    }
  })

  it('should return all fields', async () => {
    await expect(resolver(null, {}, context)).resolves.toMatchSnapshot()
  })

  it('should handle errors', async () => {
    expect.hasAssertions()
    context.getByContentType.mockRejectedValue('err')

    await expect(resolver(null, {}, context)).rejects.toThrowError('err')
  })
})
