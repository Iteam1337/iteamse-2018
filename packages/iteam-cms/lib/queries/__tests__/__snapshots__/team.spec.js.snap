// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries/team should return all fields 1`] = `
Array [
  Object {
    "avatar": "//images.ctfassets.net/rj4r6yfcesw5/2uQtyFGphOCg4U6e2mm2oY/c433294b7e3f0dfe9becd87c1c9907d3/hrn.jpg",
    "email": "hans.rollman@iteam.se",
    "headerImage": undefined,
    "location": "Stockholm",
    "name": "Hans Rollman",
    "short": "hrn",
    "title": "Samarbete",
  },
  Object {
    "avatar": "//images.ctfassets.net/rj4r6yfcesw5/64biXO1wJ2MA6wqQqWsI4o/2e3876889e13a9a1750c789b2b1e2a07/jmn.jpg",
    "email": "johanna.grahn@iteam.se",
    "headerImage": undefined,
    "location": "Göteborg",
    "name": "Johanna Månsson Grahn",
    "short": "jmn",
    "title": "Samarbete & rekrytering",
  },
  Object {
    "avatar": "https://www.gravatar.com/avatar/90ebdee038fac0f367bcec195c2536b9?s=300",
    "email": "maria.solscher@iteam.se",
    "headerImage": undefined,
    "location": "Stockholm",
    "name": "Maria Sölscher",
    "short": "msr",
    "title": "Rekrytering",
  },
  Object {
    "avatar": "https://www.gravatar.com/avatar/147291c6bde9bd3b18201d637917af97?s=300",
    "email": "mats.egardt@iteam.se",
    "headerImage": undefined,
    "location": "Stockholm",
    "name": "Mats Egardt",
    "short": "met",
    "title": "Drift & support",
  },
  Object {
    "avatar": "https://www.gravatar.com/avatar/08053b9b0035a1ad64dc9be5485f2cf5?s=300",
    "email": "rickard.laurin@iteam.se",
    "headerImage": undefined,
    "location": "Stockholm",
    "name": "Rickard Laurin",
    "short": "rln",
    "title": "Webbutvecklare",
  },
]
`;
